src/main/java/picard/sam/SamToFastq.java

```diff
@@ -286,21 +260,70 @@ protected int doWork() {
                 // Create this writer on-the-fly; if we find no second-of-pair reads, don't bother making a writer (or delegating,
                 // if we're interleaving).
                 final Lazy<FastqWriter> lazySecondOfPairWriter = new Lazy<>(() -> INTERLEAVE ? firstOfPairWriter : factory.newWriter(makeReadGroupFile(rg, "_2")));
-                writerMap.put(rg, new FastqWriters(firstOfPairWriter, lazySecondOfPairWriter, firstOfPairWriter));
+
+                writerMap.put(rg, new FastqWriters(firstOfPairWriter, lazySecondOfPairWriter, firstOfPairWriter, additionalWriters));
             }
         }
         return writerMap;
     }
 
+    protected void initializeAdditionalWriters() {

```

Comment:
I'm not a fan of this kind of extension, as it's unclear what a subclass is supposed to do if they subclass `SamToFastqWithTags` and want to override this method again.

Given that this change creates a direct dependency between these two classes, I think you'd be better off using `super.` to override the base class functionality. I think you can do this by overriding `doWork()` or  `generateWriters()`. E.g. you could override `doWork()` in `SamToFastqWithTags` as:
```
    protected int doWork() {
        setupTagSplitValues();
        return super.doWork();
    }
```